build/docker/protobuf.patch

```diff
@@ -161,30 +167,30 @@ index 4c087db..cd18e28 100644
          "classname", ClassName(descriptor_));
      }
    }
-@@ -760,9 +792,21 @@ void MessageGenerator::GenerateDescriptorMethods(io::Printer* printer) {
+@@ -760,9 +794,21 @@ void MessageGenerator::GenerateDescriptorMethods(io::Printer* printer) {
  void MessageGenerator::GenerateCommonBuilderMethods(io::Printer* printer) {
    printer->Print(
      "// Construct using $classname$.newBuilder()\n"
 -    "private Builder() {\n"
-+    "private final io.netty.util.Recycler.Handle handle;\n"
-+    "private Builder(io.netty.util.Recycler.Handle handle) {\n"
++    "private final io.netty.util.Recycler.Handle<Builder> handle;\n"
++    "private Builder(io.netty.util.Recycler.Handle<Builder> handle) {\n"
 +    "  this.handle = handle;\n"
      "  maybeForceBuilderInitialization();\n"
      "}\n"
 +    "private final static io.netty.util.Recycler<Builder> RECYCLER = new io.netty.util.Recycler<Builder>() {\n"
-+    "   protected Builder newObject(io.netty.util.Recycler.Handle handle) {\n"
++    "   protected Builder newObject(io.netty.util.Recycler.Handle<Builder> handle) {\n"
 +    "         return new Builder(handle);\n"
 +    "       }\n"
 +    "      };\n"
 +    "\n"
 +    " public void recycle() {\n"
 +    "          clear();\n"
-+    "          if (handle != null) {RECYCLER.recycle(this, handle);}\n"
-+    "      }\n"    
++    "          handle.recycle(this);\n"
```

Comment:
No, I've made the handle final and I've checked that is only set to null on the "default" instances for these object. The default instance is static and it's not used outside the class, is just there to be cloned when creating new builders. In any case it would be an error to recycle a static object.